src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int ARRAY_SIZE = 10;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+    private Object[] array = new Object[ARRAY_SIZE];
+    private int counter = 0;
+
+    private void resize(int minSize) {
+        int newSize = minSize + (minSize >> 1);
+        if (newSize - MAX_ARRAY_SIZE >= 0) {
```

Comment:
can we have `newSize` greater than `Integer.MAX_VALUE`?

Reply:
I'm blindly stupid, sorry. Now I understand.